### DISCUSSION TITLE
How to preload the parent model on a child model query

### DISCUSSION THREAD

**Discussion by haripriya-goswami** [adonisjs/core#2868] [Help] [answered]
The below code preload the related child model on the parent model instance. 
```
const users = await User
  .query()
  .preload('profile')

users.forEach((user) => {
  console.log(user.profile)
})
```
But I want to preload the parent model on the child model instance. Something like...
```
const profile = await Profile
  .query()
  .preload('user')
```
Is there any way to do it?

**Comment by Vishwas-verma** [comment, marked as answer]
For this you need to define relationship in both the model i.e user and profile.

**Reply by haripriya-goswami** [reply]
That I did. What I didn't do was add below properties to the @belongsTo relationship
```
@belongsTo(() => User, {
   localKey: 'id',
   foreignKey: 'userId'
})
  public user: BelongsTo<typeof User>
```
Thanks. It's working!
